src/query/sql/src/planner/binder/binder.rs

```diff
@@ -172,7 +173,7 @@ impl<'a> Binder {
             Statement::AlterUser(stmt) => self.bind_alter_user(stmt).await?,
 
             // Roles
-            Statement::ShowRoles => self.bind_rewrite_to_query(bind_context, "SELECT name, inherited_roles FROM system.roles ORDER BY name", RewriteKind::ShowRoles).await?,
+            Statement::ShowRoles => Plan::ShowRoles(Box::new(ShowRolesPlan {})),
```

Comment:
Can we add more where in the rewrite logic?

Reply:
the new SHOW ROLES statement displays some session context aware logic:

1. only displays the available roles for the current session
2. add is_default, is_current field like https://docs.snowflake.com/en/sql-reference/sql/show-roles.html

the session context related infomation may hard to refactor in the system table view :( as the system table would always display the same data without any regard about the session context.